### ISSUE TITLE
glog-0.3.5-intel-2017b Centos 7.4.1708 skylake Intel compiler infinite cycle

### ISSUE THREAD

**Issue by hajgato** [easybuilders/easybuild-easyconfigs#5913]
`glog` `0.3.5` fails to build with `intel/2017b` on Skylake.  The compiler gets into an infinite cycle. Not using `AVX-512` command set solves the problem. Might be a local problem.

Solution:
```
--- glog-0.3.5/Makefile.in.orig	2017-05-10 13:47:23.000000000 +0200
+++ glog-0.3.5/Makefile.in	2018-02-26 10:52:51.153987941 +0100
@@ -1209,7 +1209,7 @@
 @am__fastdepCXX_FALSE@	$(AM_V_CXX@am__nodep@)$(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libglog_la_CXXFLAGS) $(CXXFLAGS) -c -o src/libglog_la-symbolize.lo `test -f 'src/symbolize.cc' || echo '$(srcdir)/'`src/symbolize.cc
 
 src/libglog_la-signalhandler.lo: src/signalhandler.cc
-@am__fastdepCXX_TRUE@	$(AM_V_CXX)$(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libglog_la_CXXFLAGS) $(CXXFLAGS) -MT src/libglog_la-signalhandler.lo -MD -MP -MF src/$(DEPDIR)/libglog_la-signalhandler.Tpo -c -o src/libglog_la-signalhandler.lo `test -f 'src/signalhandler.cc' || echo '$(srcdir)/'`src/signalhandler.cc
+@am__fastdepCXX_TRUE@	$(AM_V_CXX)$(LIBTOOL) $(AM_V_lt) --tag=CXX $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=compile $(CXX) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(libglog_la_CXXFLAGS) $(CXXFLAGS) -xCORE-AVX2 -MT src/libglog_la-signalhandler.lo -MD -MP -MF src/$(DEPDIR)/libglog_la-signalhandler.Tpo -c -o src/libglog_la-signalhandler.lo `test -f 'src/signalhandler.cc' || echo '$(srcdir)/'`src/signalhandler.cc
 @am__fastdepCXX_TRUE@	$(AM_V_at)$(am__mv) src/$(DEPDIR)/libglog_la-signalhandler.Tpo src/$(DEPDIR)/libglog_la-signalhandler.Plo
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	$(AM_V_CXX)source='src/signalhandler.cc' object='src/libglog_la-signalhandler.lo' libtool=yes @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCXX_FALSE@	DEPDIR=$(DEPDIR) $(CXXDEPMODE) $(depcomp) @AMDEPBACKSLASH@
```

**Comment by boegel** [comment]
@hajgato Can you also work around this one by forcing `-O1` rather than overriding `-xHost` with `-xCORE-AVX2` (cfr. #5843)?

The issue is mainly that injecting `-xCORE-AVX2` is not portable, while we can consider always compiling `src/signalhandler.cc` with `-O1` to dance around this.

On a related note: we are planning to only install software with `intel/2018a` on our Skylake systems, since the compiler support is a lot better. Do you have a particular reason to persist with using `intel/2017b`?

**Comment by hajgato** [comment]
@boegel based on the experiences, `-O1` will work. The reason why I try to use architecture-dependent patches, because I do believe that we will need it in a long term (for example, `TensorFlow` compiles with `AVX512`, but it does not work. (Tried on Skylake with both `foss/2017b` and `intel/2017b`, @akesandgren tried it on `KNL`s, same result.)

We stick to `intel/2017b`, because we have a heterogeneous cluster, so we try to compile the last toolchain versions for a new architecture. (it does not have too much sense to make a cluster form 21 nodes....)

**Comment by boegel** [comment]
@hajgato PR to fix this?

**Comment by hajgato** [comment]
@boegel  https://github.com/easybuilders/easybuild-easyconfigs/pull/6360

**Comment by boegel** [comment]
fixed now #6360 is merged
